### PR TITLE
Switch to using the real metadata validate function

### PR DESCRIPTION
**Pull request by brendandburns**
Ref: https://github.com/kubernetes/kubernetes/issues/17124

@thockin

### CODE REVIEW

File: pkg/apis/extensions/validation/validation.go [hunk 1]
```diff
@@ -110,11 +110,20 @@ func ValidateHorizontalPodAutoscalerStatusUpdate(controller, oldController *exte
...
 func ValidateThirdPartyResourceUpdate(update, old *extensions.ThirdPartyResource) errs.ValidationErrorList {
-	return ValidateThirdPartyResource(update)
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&update.ObjectMeta, &old.ObjectMeta).Prefix("metadata")...)
+	allErrs = append(allErrs, ValidateThirdPartyResource(update)...)
+	return allErrs
+}
+
+func ValidateThirdPartyResourceName(name string, prefix bool) (bool, string) {
+	return true, ""
```
Comment (thockin) [inline pkg/apis/extensions/validation/validation.go]: This should probably check NameIsDNSSubdomain(name, prefix)

File: pkg/apis/extensions/validation/validation.go [hunk 2]
```diff
@@ -110,11 +110,20 @@ func ValidateHorizontalPodAutoscalerStatusUpdate(controller, oldController *exte
...
 }
 
 func ValidateThirdPartyResource(obj *extensions.ThirdPartyResource) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&obj.ObjectMeta, true, ValidateThirdPartyResourceName).Prefix("metadata")...)
+
 	if len(obj.Name) == 0 {
```
Comment (thockin) [inline pkg/apis/extensions/validation/validation.go]: this is redundant now

File: pkg/apis/extensions/validation/validation.go (hunk 1 above)
Comment (brendandburns) [inline pkg/apis/extensions/validation/validation.go]: done.

File: pkg/apis/extensions/validation/validation.go (hunk 2 above)
Comment (brendandburns) [inline pkg/apis/extensions/validation/validation.go]: done.

**Comment by brendandburns** [comment]
comments addressed, ptal.

**Comment by lavalamp** [comment]
lgtm

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by piosz** [comment]
@k8s-bot e2e test this please

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**